src/Serialization/ToricGeometry.jl

```diff
@@ -1,13 +1,30 @@
 ################################################################################
 # Toric varieties
 @register_serialization_type AffineNormalToricVariety uses_id
-@register_serialization_type NormalToricVariety uses_id
 
-function save_object(s::SerializerState, ntv::NormalToricVarietyType)
-  save_object(s, ntv.polymakeNTV)
+@register_serialization_type NormalToricVariety uses_id [:cox_ring]
+
+
+function save_object(s::SerializerState, ntv::T) where T <: NormalToricVarietyType
+  attrs = attrs_list(s, T)
+
+  if !isnothing(attrs) && any([has_attribute(ntv, attr) for attr in attrs])
+    save_data_dict(s) do
+      save_attrs(s, ntv)
+      save_object(s, ntv.polymakeNTV, :pm_data)
+    end
+  else
+    save_object(s, ntv.polymakeNTV)
+  end
```

Comment:
I think it would be cleaner to always put the polymake thing behind the `pm_data` key, independent of the existence of attributes. This would reduce both `save_object` and `load_object` to a single code path.

But maybe its just easiest if (once the attribute serialization works) you split this up into a different PR regarding toric varieties and discuss it with the people involved there (aka not me)

Reply:
yes, it would make the code a lot simpler, if this is what we want to do though, it will require doing this for a lot of other types as well, maybe not all in the same pull request, but we also require upgrade scripts for each type. The reason I did it this way for now because it saves me writing the upgrade scripts for now.

Reply:
Ah, I understand. Of course, this is something that can be adapted step by step for different types

Reply:
yes, I choose a particularly annoying example to handle to start, but for any type that already uses a save_dict_data, we can just include the save_attrs function somewhere in the body and be done

Reply:
But even in these cases I think another level in the dict would quite beneficial (see https://github.com/oscar-system/Oscar.jl/pull/4020#discussion_r1721489079)